smart_open/s3.py

```diff
@@ -553,6 +561,13 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         else:
             self.close()
 
+    def __str__(self):
+        return "s3.BufferedOutputBase(bucket_name='{}', key='{}')".format(
+            self._object.bucket_name, self._object.key)
+
+    def __repr__(self):
+        return "s3.BufferedOutputBase(bucket_name='{}', key='{}')".format(
```

Comment:
Please check out [this link](https://stackoverflow.com/questions/1436703/difference-between-str-and-repr) for an explanation of the difference between str and repr.

My understanding is that with repr, the user should be able to copy-paste the result into the Python interpreter and get back a copy of the original object. This may not always be possible, but this is something we should aim for. So, please include all the parameters passed to the constructor.

Reply:
Thank you for the feedback!

I've introduced most of the suggested changes, but I'd say that this one is still somewhat incomplete. As you said, sometimes these constructor parameters are not available and here we have a similar case where the values passed to the constructor are used to create other objects and the values themselves aren't stored anywhere.

I could try digging through the implementations of these other objects to find the values, but that seems hacky. Would it be OK if these parameters were stored in SeekableBufferedInputBase or BufferedOutputBase objects as in `self._<parameter name>`?

Reply:
Sure, I think it makes sense to store them.